src/Microsoft.Performance.Toolkit.Engine/Engine.cs

```diff
@@ -598,6 +631,64 @@ public bool TryEnableCooker(DataCookerPath dataCookerPath)
             return true;
         }
 
+        /// <summary>
+        ///     Enables the given table for processing when <see cref="Process"/>
+        ///     is called.
+        /// </summary>
+        /// <param name="descriptor">
+        ///     The table to enable.
+        /// </param>
+        /// <exception cref="TableNotFoundException">
+        ///     A table cannot be found for the given <paramref name="tableDescriptor"/>.
+        /// </exception>
+        /// <exception cref="InstanceAlreadyProcessedException">
+        ///     This instance has already been processed.
+        /// </exception>
+        public void EnableTable(TableDescriptor descriptor)
+        {
+            this.ThrowIfDisposed();
+            this.ThrowIfProcessed();
+
+            if (!this.TryEnableTable(descriptor))
+            {
+                throw new TableNotFoundException(descriptor);
+            }
+        }
+
+        /// <summary>
+        ///     Attempts to enable the given cooker for processing when <see cref="Process"/>
+        ///     is called.
+        /// </summary>
+        /// <param name="descriptor">
+        ///     The table to enable.
+        /// </param>
+        /// <returns>
+        ///     <c>true</c> if the table exists and can be enabled;
+        ///     <c>false</c> otherwise. Note that <c>false</c> is
+        ///     always returned when <see cref="IsProcessed"/> is <c>true</c>.
+        /// </returns>
+        public bool TryEnableTable(TableDescriptor descriptor)
```

Comment:
>descriptor [](start = 51, length = 10)

should we Guard.NotNull(descriptor)? if so, we should also add the ArgumentNullException to the doc comments (and bubbled up to EnableTable)